tests/cephfs/cephfs_mirroring/cephfs_mirroring_utils.py

```diff
@@ -478,6 +486,102 @@ def validate_synchronization(
                 "Snap Directory Count is not matching, Synchronization failed"
             )
 
+    @retry(CommandFailed, tries=5, delay=30)
+    def get_synced_dir_count(
+        self,
+        cephfs_mirror_node,
+        source_clients,
+        fs_name,
+        daemon_name,
+    ):
+        """
+        Gets the count of synced directories from the CephFS mirror status.
+
+        Args:
+        - cephfs_mirror_node: The node where the CephFS mirror is hosted
+        - source_clients: Source clients information
+        - fs_name: The name of the file system
+        - daemon_name: The name of the daemon
+
+        Returns:
+        - int: The count of synced directories
+
+        Raises:
+        - CommandFailed: If the directory count is not found or synchronization fails
+        """
+        fsid = self.get_fsid(cephfs_mirror_node)
+        filesystem_id = self.get_filesystem_id_by_name(source_clients, fs_name)
+        asok_file = self.get_asok_file(cephfs_mirror_node, fsid, daemon_name)
+        log.info("Validate the synced dir from cephfs-mirror status")
+        log.info("Install ceph-common on cephfs-mirror node")
+        cephfs_mirror_node.exec_command(
+            sudo=True, cmd="yum install -y ceph-common --nogpgcheck"
+        )
+        log.info("Get filesystem mirror status")
+        out, _ = cephfs_mirror_node.exec_command(
+            sudo=True,
+            cmd=f"cd /var/run/ceph/{fsid}/ ; ceph --admin-daemon {asok_file} "
+            f"fs mirror status {fs_name}@{filesystem_id} -f json",
+        )
+        data = json.loads(out)
+        log.info(data)
+        dir_count = data.get("snap_dirs", {}).get("dir_count")
+        if not data.get("snap_dirs"):
+            raise CommandFailed("Unable to get the Snap Dir")
+        if dir_count:
+            log.info(f"Snap Directory Count : {dir_count}")
+            log.info("Synchronization is successful")
+        else:
+            log.error("Snap Directory Count is not matching, Synchronization failed")
```

Comment:
As we are not matching dir_count with some input param, the error message could be, "Failed to get dir_count from mirror status, synchronization failed" ?